ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/BlockResultFactory.java

```diff
@@ -95,19 +99,29 @@ public EngineGetPayloadResultV1 payloadTransactionCompleteV1(final Block block)
     return new EngineGetPayloadResultV1(block.getHeader(), txs);
   }
 
-  public EngineGetPayloadResultV2 payloadTransactionCompleteV2(final Block block) {
+  public EngineGetPayloadResultV2 payloadTransactionCompleteV2(
+      final BlockWithReceipts blockWithReceipts) {
     final List<String> txs =
-        block.getBody().getTransactions().stream()
+        blockWithReceipts.getBlock().getBody().getTransactions().stream()
             .map(TransactionEncoder::encodeOpaqueBytes)
             .map(Bytes::toHexString)
             .collect(Collectors.toList());
 
-    final long blockValue = calculateBlockValue(txs);
-    return new EngineGetPayloadResultV2(block.getHeader(), txs, Quantity.create(blockValue));
+    final long blockValue = calculateBlockValue(blockWithReceipts);
+    return new EngineGetPayloadResultV2(
+        blockWithReceipts.getHeader(), txs, Quantity.create(blockValue));
   }
 
-  private long calculateBlockValue(final List<String> ignored) {
-    return 0L;
+  private long calculateBlockValue(final BlockWithReceipts blockWithReceipts) {
```

Comment:
I'm undecided, but wondering if we should do the calculation closer to the block/transaction processing, rather than at the API level. Would that be a trivial change? 
Perhaps it will only ever be used in this API though 🤷

Reply:
We discussed moving this method to org.hyperledger.besu.ethereum.core.BlockValueCalculator

Reply:
Updated